adam/adam_6224.py

```diff
@@ -0,0 +1,628 @@
+        format='%2.4f'
+    )
+    
+    StartupValue_0 = attribute(
+        dtype='double',
+        access=AttrWriteType.READ_WRITE,
+        format='%2.4f'
+    )
+
+    StartupValue_1 = attribute(
+        dtype='double',
+        access=AttrWriteType.READ_WRITE,
+        format='%2.4f'
+    )
+
+    StartupValue_2 = attribute(
+        dtype='double',
+        access=AttrWriteType.READ_WRITE,
+        format='%2.4f'
+    )
+
+    StartupValue_3 = attribute(
+        dtype='double',
+        access=AttrWriteType.READ_WRITE,
+        format='%2.4f'
+    )
+
+    TypeCode_0 = attribute(
+        dtype='str',
+        access=AttrWriteType.READ_WRITE,
+    )
+
+    TypeCode_1 = attribute(
+        dtype='str',
+        access=AttrWriteType.READ_WRITE,
+    )
+
+    TypeCode_2 = attribute(
+        dtype='str',
+        access=AttrWriteType.READ_WRITE,
+    )
+
+    TypeCode_3 = attribute(
+        dtype='str',
+        access=AttrWriteType.READ_WRITE,
+    )
+
+    Status_0 = attribute(
+        dtype='str',
+        access=AttrWriteType.READ,
+    )
+
+    Status_1 = attribute(
+        dtype='str',
+        access=AttrWriteType.READ,
+    )
+
+    Status_2 = attribute(
+        dtype='str',
+        access=AttrWriteType.READ,
+    )
+
+    Status_3 = attribute(
+        dtype='str',
+        access=AttrWriteType.READ,
+    )
+
+    EventStatus_0 = attribute(
+        dtype='str',
+        access=AttrWriteType.READ,
+    )
+
+    EventStatus_1 = attribute(
+        dtype='str',
+        access=AttrWriteType.READ,
+    )
+
+    EventStatus_2 = attribute(
+        dtype='str',
+        access=AttrWriteType.READ,
+    )
+
+    EventStatus_3 = attribute(
+        dtype='str',
+        access=AttrWriteType.READ,
+    )
+
+    # ---------------
+    # General methods
+    # ---------------
+
+    def init_device(self):
+        Device.init_device(self)
+        # PROTECTED REGION ID(ADAM6224.init_device) ENABLED START #
+        self.set_state(DevState.ON)
+        self.set_status("ADAM-6224 enabled")
+        # PROTECTED REGION END #    //  ADAM6224.init_device
+
+    def always_executed_hook(self):
+        # PROTECTED REGION ID(ADAM6224.always_executed_hook) ENABLED START #
+        # if not self.connected_ADAM.connect():
+        #     self.set_state(DevState.ALARM)
+        #     self.set_status("Connection  error")
+        pass
+        # PROTECTED REGION END #    //  ADAM6224.always_executed_hook
+
+    def delete_device(self):
+        # PROTECTED REGION ID(ADAM6224.delete_device) ENABLED START #
+        self.connected_ADAM.close()
+        # PROTECTED REGION END #    //  ADAM6224.delete_device
+
+    # ------------------
+    # Attributes methods
+    # ------------------
+
+    # --------------------
+    # DigitalInput methods
+    # --------------------
+
+    def read_DigitalInput_0(self):
+        # PROTECTED REGION ID(ADAM6224.DigitalInput_0_read) ENABLED START #
+        tmp = self.connected_ADAM.read_coils(0, 4)
```

Comment:
I see that you're reading some data from ADAM device only in reading the `DigitalInput_0`. Why are you not reading it in all of the other methods as well?